packages/gcc11.rb

```diff
@@ -0,0 +1,392 @@
+
+  binary_url({
+    aarch64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gcc11/11.1.0_armv7l/gcc11-11.1.0-chromeos-armv7l.tar.xz',
+     armv7l: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gcc11/11.1.0_armv7l/gcc11-11.1.0-chromeos-armv7l.tar.xz',
+       i686: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gcc11/11.1.0_i686/gcc11-11.1.0-chromeos-i686.tar.xz',
+     x86_64: 'https://gitlab.com/api/v4/projects/26210301/packages/generic/gcc11/11.1.0_x86_64/gcc11-11.1.0-chromeos-x86_64.tar.xz'
+  })
+  binary_sha256({
+    aarch64: '2db5bcee6b52e393f8a42a78543dff6339ec598f0fb852ab60ac5d1e5b620d71',
+     armv7l: '2db5bcee6b52e393f8a42a78543dff6339ec598f0fb852ab60ac5d1e5b620d71',
+       i686: 'd6d23b0cab5ea3e0922e6549140f2a0a7a4190c1739e79a49ee1051c4e02bd7d',
+     x86_64: 'e0216d9a1d1eededcca9b4a34b9e1fe7e013c3f6dea8a012bc1357c3df30b77b'
+  })
+
+  depends_on 'ccache' => :build
+  depends_on 'dejagnu' => :build # for test
+  depends_on 'hashpipe' => :build
+  depends_on 'glibc' # R
+  depends_on 'gmp' # R
+  depends_on 'isl' # R
+  depends_on 'mpc' # R
+  depends_on 'mpfr' # R
+  depends_on 'libssp' # L
+  depends_on 'libgcc_s1' # L
+
+  @gcc_global_opts = '--disable-bootstrap \
+  --disable-libmpx \
+  --disable-libssp \
+  --disable-multilib \
+  --disable-werror \
+  --enable-cet=auto \
+  --enable-checking=release \
+  --enable-clocale=gnu \
+  --enable-default-pie \
+  --enable-default-ssp \
+  --enable-gnu-indirect-function \
+  --enable-gnu-unique-object \
+  --enable-host-shared \
+  --enable-lto \
+  --enable-plugin \
+  --enable-shared \
+  --enable-symvers \
+  --enable-static \
+  --enable-threads=posix \
+  --with-gcc-major-version-only \
+  --with-gmp \
+  --with-isl \
+  --with-mpc \
+  --with-mpfr \
+  --with-pic \
+  --with-system-libunwind \
+  --with-system-zlib'
+
+  @cflags = '-fPIC -pipe'
+  @cxxflags = '-fPIC -pipe'
+  @languages = 'c,c++,jit,objc,fortran,go'
+  case ARCH
+  when 'armv7l', 'aarch64'
+    @archflags = '--with-arch=armv7-a --with-float=hard --with-fpu=neon --with-tune=cortex-a15'
+  when 'x86_64'
+    @archflags = '--with-arch-64=x86-64'
+  when 'i686'
+    @archflags = '--with-arch-32=i686'
+  end
+
+  def self.preinstall
+    gcc_version = version.split('-')[0]
+    # Use full gcc path to bypass ccache
+    stdout_and_stderr, status = Open3.capture2e('bash', '-c',
+                                                "#{CREW_PREFIX}/bin/gcc -v 2>&1 | tail -1 | cut -d' ' -f3")
+    if status.success?
+      installed_gccver = stdout_and_stderr.chomp
+      # One gets "1:" with no gcc installed.
+      unless installed_gccver.to_s == '1:' ||
+             installed_gccver.to_s == 'No' ||
+             installed_gccver.to_s == 'not' ||
+             installed_gccver.to_s == 'gcc:' ||
+             installed_gccver.to_s == gcc_version.to_s
+        abort "GCC version #{installed_gccver} already installed.".lightgreen
```

Comment:
What happens if you get rid of the `2>&1`?  Does it return an error status?

Reply:
Also this...
```
    if File.exist?("#{CREW_PREFIX}/bin/gcc")
      stdout_and_stderr, status = Open3.capture2e('bash', '-c',
                                                  "#{CREW_PREFIX}/bin/gcc -dumpversion | awk '{print $1}'")
      if status.success?
        @gccver = stdout_and_stderr.chomp
        unless @gccver.to_s == 'ccache' ||
               @gccver.to_s == ''
        end
      end
      else gccver = '11'
    end
```

Reply:
It might make sense to add generic logic to crew and then set a property like `install 'strict'` or something to indicate this version cannot be installed concurrently with another version.  So the idea is you basically wouldn't need to add this logic section to every package with similar restrictions.

Reply:
Does not renaming gcc10 -> gcc version 10 and gcc11 -> gcc version 11 give us that for free?

Reply:
If they are both named gcc, then one in theory could upgrade to gcc version 10, or to gcc version 11, and it would replace the other, since crew does an uninstall before an upgrade.

Luckily now an uninstall of gcc doesn't kill crew. :)